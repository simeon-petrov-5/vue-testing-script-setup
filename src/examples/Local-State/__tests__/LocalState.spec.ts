import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import LocalState from '../LocalState.vue';

const createWrapper = () => {
  const wrapper = mount(LocalState);

  return wrapper;
};

describe('LocalState.vue', () => {
  it('should match snapshot', () => {
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.text()).toContain('Listing data from local state');
    expect(wrapper.text()).not.toContain('More info: Character links');
  });

  it('show more info', async () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).not.toContain('More info: Character links');

    const button = wrapper.find('button');
    await button.trigger('click');
    await nextTick();
    expect(wrapper.text()).toContain('More info: Character links');
  });
});
