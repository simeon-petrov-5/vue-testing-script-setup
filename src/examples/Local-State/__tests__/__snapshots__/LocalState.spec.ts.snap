// Vitest Snapshot v1

exports[`LocalState.vue > should match snapshot 1`] = `
<div
  data-v-app=""
>
  
  <h2
    class="pageTitle text-center mt-6"
  >
    Listing data from local state
  </h2>
  <section
    class="mt-10 mb-12"
  >
    <ul
      class="grid gap-x-3 grid-cols-2 lg:w-3/4"
    >
      
      <li>
        <span>
          10: 
        </span>
      </li>
      <li>
        <span>
          Close Rick-counters of the Rick Kind: 
        </span>
      </li>
      <li>
        <span>
          April 7, 2014: 
        </span>
      </li>
      <li>
        <span>
          S01E10: 
        </span>
      </li>
      <li>
        <span>
          https://rickandmortyapi.com/api/episode/10: 
        </span>
      </li>
      <li>
        <span>
          2017-11-10T12:56:34.747Z: 
        </span>
      </li>
      
    </ul>
  </section>
  <button
    class="moreInfo"
  >
    showInfo
  </button>
  <!--v-if-->
  
</div>
`;
